src/EnergyPlus/HeatBalFiniteDiffManager.cc

```diff
@@ -827,10 +827,13 @@ namespace HeatBalFiniteDiffManager {
             SurfaceFD(Surf).PhaseChangeStateOld.allocate(TotNodes + 1);
             SurfaceFD(Surf).PhaseChangeStateOldOld.allocate(TotNodes + 1);
             SurfaceFD(Surf).PhaseChangeTemperatureReverse.allocate(TotNodes + 1);
-            SurfaceFD(Surf).condMaterialActuators.allocate(state.dataConstruction->Construct(ConstrNum).TotLayers);
-            SurfaceFD(Surf).specHeatMaterialActuators.allocate(state.dataConstruction->Construct(ConstrNum).TotLayers);
+            SurfaceFD(Surf).condMaterialActuators.allocate(TotLayers);
+            SurfaceFD(Surf).specHeatMaterialActuators.allocate(TotLayers);
             SurfaceFD(Surf).condNodeReport.allocate(TotNodes + 1);
             SurfaceFD(Surf).specHeatNodeReport.allocate(TotNodes + 1);
+            SurfaceFD(Surf).heatSourceFluxMaterialActuators.allocate(TotLayers - 1);
+            SurfaceFD(Surf).heatSourceFluxLayerReport.allocate(TotLayers - 1);
+            SurfaceFD(Surf).heatSourceFluxEnergyLayerReport.allocate(TotLayers - 1);
```

Comment:
Adding the new actuator, along with minor shuffling/refactoring.

Reply:
This has nothing to do with this PR, but (to me) the only acceptable use of `auto` is right here.

```cpp
auto & surfFD(SurfaceFD(Surf));
surf.condMaterialActuators.allocate(TotLayers);
surf.condMaterialActuators.allocate(TotLayers);
...
```

This is probably the number one thing we can do for line-shortening and code readability in EnergyPlus. This is not a great example of that, but just imagine if `SurfaceFD(Surf) were actually state.dataCondFD->SurfaceFD(Surf)`.

Reply:
I started making another pass back through with this in mind after The Great State Refactor of 2020, but then pplz got grumpy. Maybe we can do some targeted application for the worst offenders.

Reply:
Really? I would have thought people would have gotten grumpy _because of_ the #GreatStateRefactor2020 #MakeEnergyPlusStateAgain and that would be a way to ungrumpify them!